omod/src/main/java/org/openmrs/module/aijar/htmlformentry/DSDSProgramSubmissionAction.java

```diff
@@ -0,0 +1,206 @@
+package org.openmrs.module.aijar.htmlformentry;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.*;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.aijar.fragment.controller.PatientSummaryFragmentController;
+import org.openmrs.module.htmlformentry.CustomFormSubmissionAction;
+import org.openmrs.module.htmlformentry.FormEntryContext;
+import org.openmrs.module.htmlformentry.FormEntryContext.Mode;
+import org.openmrs.module.htmlformentry.FormEntrySession;
+
+import java.util.*;
+
+import static org.openmrs.module.aijar.AijarConstants.GP_DSDM_CONCEPT_ID;
+import static org.openmrs.module.aijar.AijarConstants.GP_DSDM_PROGRAM_UUID_NAME;
+
+/**
+ * Enrolls patients into DSDM programs
+ */
+public class DSDSProgramSubmissionAction implements CustomFormSubmissionAction {
```

Comment:
@slubwama Add test cases for the program submission to help aiding review of the different scenarios

Reply:
@ssmusoke This is Completed.